newdle/client/src/components/creation/CreationSuccessPage.js

```diff
@@ -38,7 +38,26 @@ export default function CreationSuccessPage() {
             invite:
           </p>
         )}
-        <div className={styles['newdle-link']}>{newdle.url}</div>
+        <Input
+          className={styles['newdle-link']}
+          readOnly
+          value={newdle.url}
+          action={
+            <Popup
+              content="Copied!"
+              on="click"
+              position="top center"
+              inverted
```

Comment:
I understand that it doesn't make much difference because the copying itself is an instantaneous operation. However, the "right" way to do this would be to only display the `<Popup />` once `writeText(...)` succeeds. I'm not sure under which conditions `writeText(...)` might fail (the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText) talk of not having permission to access the clipboard), but maybe it would be worth making the `<Popup />` controlled and adding an additional state to take care of that.
@ThiefMaster opinions?

Reply:
Agreed, for showing the popup I think using state that's only set after copying would be nicer.. (and then maybe using setTimeout to hide it after a second or two).

We should probably also try..catch around the call to avoid an error when the promise does get rejected. Probably no need to actually do anything in there for now, since it seems unlikely that it happens..

Reply:
I tested in firefox with `dom.events.asyncClipboard` disabled and the button gets hidden as expected. I think for now it's fine, if we find out that there are cases where `navigator.clipboard` is defined but copying still fails we can still add error handling there